build.sbt

```diff
@@ -1,13 +1,24 @@
 import sbt._
 
+// https://github.com/sbt/sbt-git
+enablePlugins(GitVersioning)
+
 lazy val commonSettings = Seq(
+  // sbt-git provides the isSnapshot.value variable, which will return true if there
+  // are no tags associated with the HEAD commit, or if there are uncommitted
+  // changes. However, sbt-git only appends the -SNAPSHOT suffix if there are
+  // uncommitted changes in the workspace. 
+  //
+  // https://github.com/sbt/sbt-git/blob/f8caf9365be380cf101e9605af159b5e7f842d0c/src/main/scala/com/typesafe/sbt/SbtGit.scala#L173
+  version := {
+    // Avoid Cyclic reference involving error

```

Comment:
I'm looking for all non tagged commits to have a `-SNAPSHOT` suffix on the version (the behavior of the boolean `isSnapshot`). 

For some reason, sbt-git only appends the `-SNAPSHOT` suffix if there are uncommitted changes in the workspace, regardless of the boolean value of `isSnapshot`.

typelevel/cats and Raster Foundry take the approach of [using a command alias](https://github.com/typelevel/cats/blob/96a2b1629ef082c3e7b6086845e1e955b25aa505/build.sbt#L757) to append the `-SNAPSHOT` suffix. 

The problem with this is, the logic for publishing releases and snapshots needs to be separate. I wanted a clean approach that would allow us to use the same `publish` command with sbt for both releases and snapshots.

The logic below should be the equivalent of the command alias linked above. 

I modeled this after: https://github.com/ScorexFoundation/sigmastate-interpreter/blob/5f01afb63c211994c3088a7455058f2f29a62f26/build.sbt#L33

Reply:
Just wanted to surface that we might also be able to deal with this conditional via multiple `deploy` blocks in `.travis.yml` (e.g., one command exclusively for tagged commits and another for `develop`).